content/blog/prototyping/index.md

```diff
@@ -0,0 +1,767 @@
+You can make the rough edges obvious so that you can sort them out later.
+Don't let perfect be the enemy of good.
+
+One of the biggest mistakes I observe is an engineer's perfectionist instinct to jump on minor details which don't have a broad enough impact to warrant the effort.
+It's better to have a working prototype with a few rough edges than a perfect implementation of a small part of the system.
+
+Remember: you are exploring!
+Use a coarse brush to paint the landscape first.
+Try to get into a flow state where you can quickly iterate.
+Don't get distracted by the details too early.
+During this phase, it's also fine to throw away a lot failed attempts. 
+
+### Use Simple Types
+
+Even while prototyping, the type system is not going away.
+There are ways to make this a blessing rather than a curse.
+
+Use simple types like `i32`, `String`, `Vec` in the beginning.
+We can always make things more complex later if we have to -- the reverse is much harder.
+
+Here's a quick reference for common prototype-to-production type transitions:
+
+| Prototype        | Production          | When to switch                                                                     |
+| ---------------- | ------------------- | ---------------------------------------------------------------------------------- |
+| `String`         | `&str`              | When you need to avoid allocations or store string data with a clear lifetime      |
+| `Vec<T>`         | `&[T]`              | When the owned vector becomes too expensive to clone or you can't afford the heap  |
+| `Box<T>`         | `&T` or `&mut T`    | When `Box` becomes a bottleneck or you don't want to deal with heap allocations    | 
+| `Rc<T>`          | `&T`                | When the reference counting overhead becomes too expensive or you need mutability  | 
+| `Arc<Mutex<T>>`  | `&mut T`            | When you can guarantee exclusive access and don't need thread safety               |
+
+These owned types sidestep most ownership and lifetime issues, but they do it by allocating memory on the heap - just like Python or JavaScript would.
+
+You can always refactor when you actually need the performance or tighter resource usage, but chances are you won't.
+
+### Make use of type inference
+
+Rust is a statically, strongly typed language.
+It would be a deal-breaker to write out all the types all the time if it weren't for Rust's type inference.
+
+You can often omit (also called "elide") the types and let the compiler figure it out from the context.
+
+```rust
+let x = 42;
+let y = "hello";
+let z = vec![1, 2, 3];
+```
+
+This is a great way to get started quickly and defer the decision about types to later.
+The system scales well with more complex types, so you can use this technique even in larger projects.
+
+```rust
+let x: Vec<i32> = vec![1, 2, 3];
+let y: Vec<i32> = vec![4, 5, 6];
+
+// Rust can infer the item type of z
+let z = x.iter().chain(y.iter()).collect::<Vec<_>>();
```

Comment:
The bit about misconceptions by new Rust developers seems to imply, that this article is targeted at them. _I_ totally understand, that what you're referring to here as being inferred is the `_` bit, and that the `Vec` part is necessary for the `collect` and doesn't detract from your point. Will they?

Maybe flattening and creating a sum (with `for` or iterator adapters) could be a better example?